src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -343,6 +343,9 @@ private static int GetMinRequiredLength(Regex r)
         [InlineData("(?:a{1,2}){4}", "a{4,8}")]
         // Nested atomic
         [InlineData("(?>(?>(?>(?>abc*))))", "(?>ab(?>c*))")]
+        [InlineData("(?>(?>(?>(?>))))", "")]
```

Comment:
people who write bizarre, redundant patterns are going to be pleased by this release :)

Reply:
Heh, yeah.  I've stayed away from spending any time optimizing for things that are utterly useless, e.g. I contemplated doing a pass over a Concatenate looking for any children that are Nothing, in which case the whole Concatenate could be replaced by a Nothing, but for that to happen a dev would need to write something like `abc(?!)def`, and why would they do that?  Instead I've focused on patterns I've seen actually show up, either because they're reasonable to write or because something reasonable to write reduced through our various other optimizations down to something a dev wouldn't necessarily write.